src/main/java/tech/gruppone/stalker/server/services/UserService.java

```diff
@@ -61,4 +63,22 @@
                         .build()))
         .then();
   }
+
+  public Mono<Void> putUserById(UserDataDto userDataDto, Long userId) {
+    return userRepository
+        .findById(userId)
+        .filter(userDao -> userDao.getEmail().equals(userDataDto.getEmail()))
+        .switchIfEmpty(Mono.error(new NotFoundException()))
+        .and(
+            userDataRepository.save(
+                UserDataDao.builder()
+                    .userId(userId)
+                    .firstName(userDataDto.getFirstName())
+                    .lastName(userDataDto.getLastName())
+                    .birthDate(userDataDto.getBirthDate())
+                    .createdDate(LocalDateTime.now())
+                    .lastModifiedDate(LocalDateTime.now())
+                    .build()))
```

Comment:
Attenzione ad una cosa qui: è indifferente che filtro la mail che ricevo dalla query (passando lo userId) confrontandola con mail inserita nel requestBody, in quanto **and** esegue comunque il codice dopo lo **switchIfEmpty**.

Questo vuol dire che qualsiasi sia la mail nel requestBody, io posso modificare i miei dati

Reply:
Detto onestamente non sapevo come funzionava. Devo cambiare la funzione